### DISCUSSION TITLE
Add option to resolve a hostname to a specific IP address

### DISCUSSION THREAD

**Discussion by borekb** [Kong/insomnia#3832] [Ideas]
For example, I'd like to test a GET request to `subdomain.localhost` which browsers resolve to 127.0.0.1 but most operating systems don't.

Curl has a convenient `--resolve` option for these situations, for example, compare plain request:

```console
$ curl -I http://subdomain.localhost
curl: (6) Could not resolve host: subdomain.localhost
```

vs. adding `--resolve`:

```console
$ curl -I --resolve subdomain.localhost:80:127.0.0.1  http://subdomain.localhost
HTTP/1.1 200 OK
Vary: Origin
...
```

It would be great to a similar functionality in Insomnia. As a workaround, it's possible to add this to `/etc/hosts` but I think it's better when a higher-level tool like curl or Insomnia support this directly.

**Comment by dimitropoulos** [comment]
does using https://readme.localtest.me/ work for you, at all?  I've used it before with insomnia in a similar situation.

**Reply by evelynhathaway** [reply]
Localtest.me doesn't appear to help with local HTTPS or SSL debugging. Setting manual resolve rules are still very helpful for developing and debugging. Opening and configuring `/etc/hosts` works but is far from ideal.

## SSL and SNI Debugging

I was recently trying to verify that my server was sending a valid certificate. I couldn't test how it responds to a request using the SNI of `example.com`. My use case was for verifying my upstream server could respond with the correct certificate for Cloudflare with SSL strict mode turned on.

This also helps with debugging virtual hosts, servers behind a proxies and CDNs.

## Local Development and HTTPS

I cannot develop against localtest.me for testing against `[service].mycompany.com` since I cannot get a valid certificate for localtest.me, cross origin resources will not be accessible since they are whitelisted to `*.mycompany.com`, cookies will not be accessible since they are set on `Domain=mycompany.com`.

**Comment by ecerulm** [comment]
Having this a "resolve" option akin to "curl --resolve" would be really good also for testing in kubernetes minikube.  

Usually in local development you want to have an Ingress for www.mycompany.com but  then you need to connect to `curl --resolve www.mycompany.com:80:127.0.0.01 www.company.com` so that the `Host:  www.mycompany.com`. 

Obviously editing /etc/hosts also works but it will be more convenient if you can set it the UI, just like you can provide `--resolve xxx` options to curl .

**Comment by NickRimmer** [comment]
What if call the IP address with Host header?

![image](https://github.com/Kong/insomnia/assets/1754265/a1f6aa59-13eb-4581-bbb4-5d43a54f90d1)

**Reply by ecerulm** [reply]
Yes, that works, I've used that as well, but again I think it's better with an resolve (so that I can switch between calling my dev  and the real one). Right now I can do that but I have to change the /etc/hosts every time I switch. For me I would be more convenient if I can switch enable/disable a "resolve" option somewhere in the UI

**Reply by NickRimmer** [reply]
You can set it as variables from environments per stage, like for dev and prod use different IP addresses or Host names

**Reply by NickRimmer** [reply]
It is much easier for me to have all environment/stage-related settings in one place, instead of an additional list of hostname resolving.
